Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-15.ca-central-1.compute.internal
Sha: 14f508f83e9e6ebed860a814954712097d973a6d
Number of Error(s): 4
/home/centos/smoketest/PR-13927/HPCC-Platform/tools/hidl/hidlcomp.cpp:4366:46: error: \xe2\x80\x98strcmpi\xe2\x80\x99 was not declared in this scope
make[2]: *** [tools/hidl/CMakeFiles/hidl.dir/build.make:126: tools/hidl/CMakeFiles/hidl.dir/hidlcomp.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3740: tools/hidl/CMakeFiles/hidl.dir/all] Error 2
make: *** [Makefile:163: all] Error 2

[Build error:](https://gist.github.com/c855188c4e7d9042f0d89cdc0faa3b40)
Build log [Build log](https://gist.github.com/HPCCSmoketest/c855188c4e7d9042f0d89cdc0faa3b40#file-result-20-07-01-17-52-12-log)

--- a/tools/hidl/hidlcomp.cpp
+++ b/tools/hidl/hidlcomp.cpp
@@ -1898,7 +1898,7 @@ bool ParamInfo::write_mapinfo_check(int indents, const char* ctxvar)
     return hasOutput;
 }
 
-void ParamInfo::write_esp_marshall(bool isRpc, bool encodeXml, bool checkVer, int indents)
+void ParamInfo::write_esp_marshall(bool isRpc, bool encodeXml, bool checkVer, int indents, bool encodeJson)
 {
     const char *soap_path=getMetaString("soap_path", NULL);
     char *path = (soap_path!=NULL) ? strdup(soap_path) : NULL;
@@ -1969,7 +1969,7 @@ void ParamInfo::write_esp_marshall(bool isRpc, bool encodeXml, bool checkVer, in
             else
             {
                 outf("%s, \"\", %s", encode, prefix);
-                if (getMetaInt("json_inline"))
+                if (getMetaInt("json_inline", !encodeJson))
                     outs(", false");
                 outs(");\n");
             }
@@ -4359,10 +4359,21 @@ void EspMessageInfo::write_esp()
 
         if (hasMapInfo())
             outf("\t\tdouble clientVer = ctx ? ctx->getClientVersion() : -1;\n");
+
+        bool encodeJSON = true;
+        const char * name = getName();
+        unsigned nameLength = strlen(name);
+        if (nameLength >= 2)
+        {
+            const char * nameEnding = &name[nameLength - 2];
+            bool isResponseEx = strcmp(nameEnding, "Ex") == 0;
+            encodeJSON = !(isResponseEx && (stricmp(getParams()->getXmlTag(), "Response") == 0));
+        }
+
         for (pi=getParams();pi!=NULL;pi=pi->next)
         {
             if (!pi->getMetaInt("attribute"))
-                pi->write_esp_marshall(false, encodeXML, true, 2);
+                pi->write_esp_marshall(false, encodeXML, true, 2, encodeJSON);
         }
         outs("\t}\n");
     }

--- a/tools/hidl/hidlcomp.h
+++ b/tools/hidl/hidlcomp.h
@@ -443,7 +443,7 @@ public:
     void write_esp_declaration();
     void write_esp_ng_declaration(int pos);
     void write_esp_init(bool &isFirst, bool removeNil);
-    void write_esp_marshall(bool isRpc, bool encodeXml, bool checkVer=false, int indent=1);
+    void write_esp_marshall(bool isRpc, bool encodeXml, bool checkVer=false, int indent=1, bool encodeJson=true);
     void write_esp_unmarshall(const char *rpcvar, bool useBasePath=false, int indents=1);
     void write_esp_unmarshall_properties(const char *propvar, const char *attachvar, int indents=1);
     void write_esp_unmarshall_soapval(const char *var, int indents=1);

